include/rcutils/snprintf.h

```diff
@@ -20,30 +20,41 @@ extern "C"
 {
 #endif
 
-#include <stdio.h>
+#include <stdarg.h>
+#include <stddef.h>
+
+#include "rcutils/macros.h"
+#include "rcutils/visibility_control.h"
 
 /// Format a string.
 /**
  * This function just wraps snprintf() as defined in C11 in a portable way.
  *
- * On Windows this defaults to the _TRUNCATE behavior of _snprintf_s().
+ * On Windows this defaults to the _TRUNCATE behavior of _snprintf_s(), but
+ * only returns -1 if errno is not 0.
+ * Unlike _snprintf_s() which returns -1 when truncation occurs, this function
+ * behaves like snprintf() (http://en.cppreference.com/w/cpp/io/c/fprintf):
+ *
+ * > Number of characters written if successful or negative value if an error
+ * > occurred.
+ * > If the resulting string gets truncated due to buf_size limit, function
+ * > returns the total number of characters (not including the terminating
+ * > null-byte) which would have been written, if the limit was not imposed.
  *
  * \see snprintf()
+ * \see _snprintf_s()
+ * \returns -1 if there is an error, but not if there is truncation
```

Comment:
`vsnprintf` can return a "negative value if an error occurred", so we should either map that to -1 or use the same wording here